packages/components/src/Drawer/Drawer.css

```diff
@@ -48,10 +48,21 @@
 
 .header {
   display: flex;
-  padding: var(--space-base) var(--space-base) 0 var(--drawer--base-padding);
+  padding: var(--drawer--base-padding) var(--drawer--base-padding) 0
+    var(--drawer--base-padding);
   background-color: transparent;
   justify-content: space-between;
-  align-items: center;
+  align-items: flex-start;
+  gap: var(--space-base);
+}
+
+.heading {
```

Comment:
Do we need `overflow: hidden` in `heading` or something similar?

If I have a long word, the dismiss icon goes beyond the page at a smaller breakpoint/when the drawer snaps to a smaller size.
![Screenshot 2024-09-20 at 5 04 39 PM](https://github.com/user-attachments/assets/f5e6bfec-e12f-46a3-80a0-6558072ed57f)

SideDrawer seems to handle it a bit better

![Screenshot 2024-09-20 at 5 16 34 PM](https://github.com/user-attachments/assets/f94c2f7e-138d-4592-b57c-6227bd3cb239)

Reply:
@taylorvnoj I'll see if I can fix that, sometimes limited CSS solutions to really long words when their containers are smaller.

Reply:
@taylorvnoj played with a few things but I don't believe this will be fixable without some potential unpredictability in where words break. We could do something like `word-break: break-word` which will work in some cases but it doesn't have a predictable pattern on where it breaks the words, which is usually an accessibility concern for a screen reader.

I'd say we don't solve it given how many drawers have titles set by us. In cases where drawer titles are set by the user, I imagine it would be very uncommon to have words that exceed the width of the drawer.